mcs/class/System/Internal.Cryptography/OidLookup.Managed.cs

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright (c) 2018 Microsoft
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+	/// <summary>
+	/// Fully managed implementation of OidLookup for the commonly supported Oids.
+	/// </summary>
+	internal static partial class OidLookup
```

Comment:
This needs to be in the same namespace as the partial class in CoreFX. It lists the OIDs that are missing in OidLookup.cs in CoreFX compared to Mono prior to this patch. It may be worth it to move the missing oids to the generic OidLookup and submit that as PR to CoreFX, which would at the same time render their OidLookup.OSX.cs obsolete.

Reply:
Apparently the Windows names that are preserved in OidLookup.cs are only those that are not localized in different languages. The expectation is that if the names could differ on Windows between languages they could also differ between platforms. So, short term it makes some sense to keep a Mono-specific implementation. Long term it makes sense to switch to the OidLookup.[OSX/Unix/Windows].cs along with the full PAL layer and remove the unit tests that depend on the names. Thoughs?